pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -209,17 +197,6 @@ func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidat
 	log := d.Log.WithValues("name", name, "method", method)
 	ctx = klog.NewContext(ctx, log)
 
-	startStorage := time.Now()
-	deletedS, async, err := d.Storage.Delete(ctx, name, deleteValidation, options)
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			log.WithValues("objectList", deletedS).Error(err, "could not delete from duplicate storage")
-			d.recordStorageDuration(true, mode2Str, d.resource, method, startStorage)
-		}
-		return deletedS, async, err
-	}
-	d.recordStorageDuration(false, mode2Str, d.resource, method, startStorage)
```

Comment:
This was the only function calling unified first and then legacy. I changed the order so we can expect the same execution order from all the functions